src/main/java/org/openstreetmap/atlas/geography/atlas/sqlite/SQLiteItemWriter.java

```diff
@@ -0,0 +1,216 @@
+package org.openstreetmap.atlas.geography.atlas.sqlite;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Optional;
+
+import org.openstreetmap.atlas.exception.CoreException;
+import org.openstreetmap.atlas.geography.Location;
+import org.openstreetmap.atlas.geography.Rectangle;
+import org.openstreetmap.atlas.geography.atlas.items.Area;
+import org.openstreetmap.atlas.geography.atlas.items.AtlasItem;
+import org.openstreetmap.atlas.geography.atlas.items.Edge;
+import org.openstreetmap.atlas.geography.atlas.items.ItemType;
+import org.openstreetmap.atlas.geography.atlas.items.Line;
+import org.openstreetmap.atlas.geography.atlas.items.Node;
+import org.openstreetmap.atlas.geography.atlas.items.Point;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedAtlas;
+import org.openstreetmap.atlas.geography.atlas.packed.PackedTagStore;
+import org.openstreetmap.atlas.geography.converters.WkbLocationConverter;
+import org.openstreetmap.atlas.utilities.maps.LongToLongMap;
+
+/**
+ * To save all of the data within an atlas, we need to get each {@link AtlasItem} type and iterate
+ * though them with inserts into the database. This class declares that a given writer needs to get
+ * the atlas to do its job.
+ *
+ * @author hallahan
+ * @author lcram
+ */
+public class SQLiteItemWriter
+{
+    private static final String INSERT_SQL = "INSERT INTO %s VALUES (?, ?, ?, ?)";
+    private static final String INSERT_TAGS_SQL = "INSERT INTO %s_tags VALUES (?, ?, ?)";
+    private static final String INSERT_RTREE_SQL = "INSERT INTO %s_rtree VALUES (?, ?, ?, ?, ?)";
+
+    // database column constants
+    private static final int FIRST_COLUMN = 1;
+    private static final int SECOND_COLUMN = 2;
+    private static final int THIRD_COLUMN = 3;
+    private static final int FOURTH_COLUMN = 4;
+    private static final int FIFTH_COLUMN = 5;
+
+    private final Connection connection;
+    private final PreparedStatement itemStatement;
+    private final PreparedStatement tagsStatement;
+    private final Optional<PreparedStatement> rtreeStatementOptional;
+
+    private final LongToLongMap identifierToTagStoreIndex;
+    private final PackedTagStore tagStore;
+    private final ItemType itemType;
+
+    SQLiteItemWriter(final Connection connection, final boolean buildIndices,
+            final PackedTagStore tagStore, final LongToLongMap identifierToTagStoreIndex,
+            final ItemType itemType) throws SQLException
+    {
+        this.tagStore = tagStore;
+        this.identifierToTagStoreIndex = identifierToTagStoreIndex;
+        this.itemType = itemType;
+        this.connection = connection;
```

Comment:
Again generally not best practice to store connections as class variables.

Reply:
In the past I've done this, but for these purposes, we know we only have one connection to the DB ever. Explicitly closing the connection has no effect on the state of the underlying database file.